service/src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -901,6 +901,36 @@ public String getOrCreatePetSaEmail(String projectId, AuthenticatedUserRequest u
     }
   }
 
+  /**
+   * Get a pet service account token for a given project and user request
+   *
+   * @param projectId - GCP project ID
+   * @param userRequest - original user request
+   * @return - pet service account token
+   */
+  public String getOrCreatePetSaToken(String projectId, AuthenticatedUserRequest userRequest) {
+    GoogleApi googleApi = samGoogleApi(userRequest.getRequiredToken());
+    try {
+      return googleApi.getPetServiceAccountToken(projectId, new ArrayList<>(SAM_OAUTH_SCOPES));
```

Comment:
This is the part I'm not sure about. Do we need any more scopes for the pet than these?

Reply:
Besides the possibility of having the token expire, I thought the reason we wanted to use the pet was to get a scope that would allow us to use the data transfer service. I would expect the tests to fail if you need more scopes.

Reply:
If you're going to use the pet SA token to talk to GCP, then you'll need the `cloud-platform` OAuth scope.